### PR TITLE
PICARD-319: Add an action toolbar to file browser

### PR DESCRIPTION
**Pull request by samj1912**

### CODE REVIEW

**Comment by samj1912** [comment]
I want to add a couple more buttons to this toolbar 

1. A set home/set starting directory button
2. A move files here button
3. Maybe a toggle hidden files button?

**Comment by zas** [comment]
It would be nice to provide a screenshot for such UI changes in the PR description.

File: picard/ui/filebrowser.py [hunk 1]
```diff
@@ -23,7 +23,51 @@
...
+        paths = self._selected_paths()
+        files = []
+        directories = []
+        for path in paths:
+            if os.path.isfile(path):
+                files.append(path)
+            else:
+                directories.append(path)
+
+        if len(files):
```
Comment (zas) [inline picard/ui/filebrowser.py]: `if files:` should be enough.

File: picard/ui/filebrowser.py [hunk 2]
```diff
@@ -23,7 +23,51 @@
...
+            self.tagger.add_files(files)
+        if len(directories):
```
Comment (zas) [inline picard/ui/filebrowser.py]: `if directories:` should be enough

**Comment by samj1912** [comment]
Screenshot

Screenshot of the same. I plan to add 3 more buttons to that toolbar

**Comment by zas** [comment]
> Screenshot of the same. I plan to add 3 more buttons to that toolbar

Imho this should be discussed, i'm not an UI guy at all, but i have the feeling adding more buttons don't make things easier.
I'll all for changes in the UI. What about creating a thread about this on forums so non-devs can participate ?

**Comment by samj1912** [comment]
I modified the code to show text labels under icons, but I don't think that is meant for the filebrowser toolbar.

filebrowser


This is how it looks and I think it kind of defeats the purpose of having a small toolbar. Also the '+' sign is intuitive enough and I have tooltips for more info.

**Comment by mineo** [comment]
The + sign is intuitive for the single add-items-to-picard function, but if more buttons are going to be added, not all of them are going to have an obvious meaning. We should take this into consideration because either more buttons are going to be added or the Add one is going to stay the only one.
In the latter case, having it just say "Add" and use the whole width of the toolbar instead of being a lonely icon on the left might be preferrable.

**Comment by samj1912** [comment]
I am thinking of "Add new folder" button. Which I think will be intuitive as it is used a lot. and a move files here icon like this? move files

File: picard/ui/filebrowser.py [hunk 3]
```diff
@@ -23,7 +23,51 @@
...
+
+    def add_items(self):
+        paths = self._selected_paths()
+        files = []
+        directories = []
+        for path in paths:
+            if os.path.isfile(path):
+                files.append(path)
+            else:
+                directories.append(path)
```
Comment (zas) [inline picard/ui/filebrowser.py]: Why not add directories directly here ?

I understand why you are building a list of files to pass to `tagger.add_files()`, but you may skip this for directories and just do `self.tagger.add_directory(path)` here.

File: picard/ui/filebrowser.py [hunk 4]
```diff
@@ -23,7 +23,51 @@
...
+        self.file_browser.save_state()
+
+    def restore_state(self):
+        self.file_browser.restore_state()
+
+    def _selected_paths(self):
+        paths = []
+        for index in self.file_browser.selectedIndexes():
+            paths.append(self.file_browser.model.filePath(index))
+        return paths
```
Comment (zas) [inline picard/ui/filebrowser.py]: A list comprehension is prolly faster:
```python
return [self.file_browser.model.filePath(index) for index in self.file_browser.selectedIndexes()]
```

**Comment by samj1912** [comment]
image

Updated look for add items button. I think we can add other buttons after more discussion.

**Comment by samj1912** [comment]
This PR is complete from my side pending any reviews :)

**Comment by mineo** [comment]
To be honest, I still can't get myself to like that toolbar very much :( On your screenshot it looks really good, but on a higher resolution the toolbar is 70%+ empty whitespace. Maybe this is just because there's only one button now there at the moment...

Since I don't have any other suggestions, I'll leave it to zas to decide about this.

**Comment by zas** [comment]
> To be honest, I still can't get myself to like that toolbar very much :( On your screenshot it looks really good, but on a higher resolution the toolbar is 70%+ empty whitespace. Maybe this is just because there's only one button now there at the moment...

I have the same feeling. I think we can do better, not sure yet how.

**Comment by mineo** [comment]
Let's postpone this until we have more operations that we'd like to put into that toolbar.

**Pull request closed without merging**